src/db/add_package.rs

```diff
@@ -213,6 +217,35 @@ fn convert_dependencies(pkg: &MetadataPackage) -> Vec<(String, String, String)>
         .collect()
 }
 
+/// Reads features and converts them to Vec<Feature> with default being first
+fn get_features(pkg: &MetadataPackage) -> Vec<Feature> {
+    let mut features = Vec::with_capacity(pkg.features.len());
+    if let Some(subfeatures) = pkg.features.get("default") {
+        features.push(Feature::new("default".into(), subfeatures.clone()));
+    };
+    features.extend(
+        pkg.features
+            .iter()
+            .filter(|(name, _)| !name.eq(&"default"))
+            .map(|(name, subfeatures)| Feature::new(name.clone(), subfeatures.clone())),
+    );
+    features.extend(get_optional_features(pkg));
+    features
+}
+
+fn get_optional_features(pkg: &MetadataPackage) -> Vec<Feature> {
+    pkg.dependencies
+        .iter()
+        .filter(|dep| dep.optional)
+        .filter(|dep| {
+            pkg.features
+                .values()
+                .any(|features| features.iter().any(|sub| sub.eq(&dep.name)))
```

Comment:
You can't have a feature with the same name as a dependency.

Reply:
We probably want to not filter anything out when storing the features into the database, and instead filter them during display so that we can change the heuristics and have them apply to existing data.